test/e2e/features/install.feature

```diff
@@ -0,0 +1,299 @@
+    When ClusterExtension is applied
+      """
+      apiVersion: olm.operatorframework.io/v1
+      kind: ClusterExtension
+      metadata:
+        name: ${NAME}
+      spec:
+        namespace: ${TEST_NAMESPACE}
+        serviceAccount:
+          name: olm-sa
+        source:
+          sourceType: Catalog
+          catalog:
+            packageName: <package-name>
+            selector:
+              matchLabels:
+                "olm.operatorframework.io/metadata.name": test-catalog
+      """
+    Then ClusterExtension is rolled out
+    And ClusterExtension is available
+    And bundle "<package-name>-operator.1.2.0" is installed in version "1.2.0"
+    And resource "networkpolicy/test-operator-network-policy" is installed
+    And resource "configmap/test-configmap" is installed
+    And resource "deployment/test-operator" is installed
+
+    Examples:
+      | package-name  |
+      | test-mirrored |
+      | dynamic       |
+
+
+  Scenario: Report that bundle cannot be installed when it exists in multiple catalogs with same priority
+    Given ClusterCatalog "extra" serves bundles
+    When ClusterExtension is applied
+      """
+      apiVersion: olm.operatorframework.io/v1
+      kind: ClusterExtension
+      metadata:
+        name: ${NAME}
+      spec:
+        namespace: ${TEST_NAMESPACE}
+        serviceAccount:
+          name: olm-sa
+        source:
+          sourceType: Catalog
+          catalog:
+            packageName: test
+      """
+    Then ClusterExtension reports Progressing as True with Reason Retrying and Message:
+      """
+      found bundles for package "test" in multiple catalogs with the same priority [extra-catalog test-catalog]
+      """
+
+  @SingleOwnNamespaceInstallSupport
+  Scenario: watchNamespace config is required for extension supporting single namespace
+    Given service account "olm-admin" in test namespace is cluster admin
+    And resource is applied
+      """
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: single-namespace-operator-target
+      """
+    And ClusterExtension is applied
+      """
+      apiVersion: olm.operatorframework.io/v1
+      kind: ClusterExtension
+      metadata:
+        name: ${NAME}
+      spec:
+        namespace: ${TEST_NAMESPACE}
+        serviceAccount:
+          name: olm-admin
+        source:
+          sourceType: Catalog
+          catalog:
+            packageName: single-namespace-operator
+            selector:
+              matchLabels:
+                "olm.operatorframework.io/metadata.name": test-catalog
+      """
+    And ClusterExtension reports Progressing as True with Reason Retrying and Message:
+      """
+      error for resolved bundle "single-namespace-operator.1.0.0" with version "1.0.0":
+      invalid ClusterExtension configuration: invalid configuration: required field "watchNamespace" is missing
+      """
+    When ClusterExtension is updated to set config.watchNamespace field
```

Comment:
Future enhancement, patching of just the modified fields?

Reply:
I would do in followup PR, if you do not mind.